52.md

```diff
@@ -0,0 +1,142 @@
+
+This NIP is intentionally not defining who is authorized to attend a calendar event if the user who RSVP'd has not been tagged. It is up to the calendar event creator to determine the semantics.
+
+This NIP is also intentionally not defining what happens if a calendar event changes after an RSVP is submitted.
+
+### Format
+
+The format uses a parameterized replaceable event kind `31924` with a list of tags as described below:
+* `a` (required) reference tag to kind `31923` calendar event being responded to per [NIP-33](33.md)
+* `d` (required) universally unique identifier. Generated by the client creating the calendar event RSVP.
+* `t` (required) attendance status. It can be one of the following values: `yes`, `no`, `maybe`
+* `t` (optional) availability status. It can be one of the following values: `free`, `busy`
+* `content` (optional) free-form note that adds more context to this calendar event response.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <Unix timestamp in seconds>,
+  "kind": "31924",
+  "tags": [
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["d", "<UUID>"]
+
+    ["t", "<yes/no/maybe>"],
+    ["t", "<free/busy>"],
+
+    ["content", "<note>"]
+  ]
+}
+```
+
+## Calendar
+
+A calendar is a collection of calendar events, represented as a custom replaceable list event using kind `31925`. A user can have multiple calendars. One may create a calendar to segment calendar events for specific purposes. e.g., personal, work, travel, meetups, and conferences.
+
+### Format
+
+The format uses a custom replaceable list of kind `31925` with a list of tags as described below:
+* `d` (required) calendar name
+* `a` (repeated) reference tag to kind `31923` calendar event being responded to per [NIP-33](33.md)
+
+```json
+{
+  "kind": 31925,
+  "tags": [
+    ["d", "<calendar name>"],
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"]
+  ]
+}
+```
+
+## Unsolved Limitations
+
+* No private events
+
+## Intentionally Unsupported Scenarios
+
+### Recurring Calendar Events
+
+Recurring calendar events come with a lot of complexity, making it difficult for software and humans to deal with. This complexity includes time zone differences between invitees, daylight savings, leap years, multiple calendar systems, one-off changes in schedule or other metadata, etc.
+
+This NIP intentionally omits support for recurring calendar events and pushes that complexity up to clients to manually implement if they desire. i.e., individual calendar events with duplicated metadata represent recurring calendar events.
+
+### Time Zones
```

Comment:
This is a mistake. Timezones matter, and we can't simply ignore their existence. What's more, timezones as unfun as they can be, are a solved problem.

Reply:
Timestamps should be a nice universal way to solve that without doing too much at the base protocol.

The client can convert that timestamp into whatever the local time is set to

Reply:
There is no need for anything extra. 

The UNIX timestamp we use is on GMT. If you want to schedule an appointment, apps need to convert from the UI (say, EST) to GMT and send it. When receiving, the UI parses the event and converts back to the timezone of the receiver. If you are traveling, the UI should have a timezone selector so you can mark the event at 5pm JPN time, for instance. The UI converts from JPN to GMT and sends the event.

Reply:
Ended up adding an optional start and end time zone identifier in 2ce758a5be60d7fbda142192b87b57ce552aee50

Use case: I'm in North America and would like to know the schedule of Nostrasia in Tokyo before I get there in Asia/Tokyo time, but I don't want to need to compute it from the geohash as it's non-trivial and implicit.